File: Modules/School/Grade/GradeRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace ExamAppSerene.School;

[ConnectionKey("Default"), Module("School"), TableName("[exam].[Grade]")]
[DisplayName("Grade"), InstanceName("Grade")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class GradeRow : Row<GradeRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Grade Id"), Identity, IdProperty]
    public int? GradeId
    {
        get => fields.GradeId[this];
        set => fields.GradeId[this] = value;
    }

    [DisplayName("Code"), Size(255), NotNull, QuickSearch, NameProperty]
    public string Code
    {
        get => fields.Code[this];
        set => fields.Code[this] = value;
    }

    [DisplayName("Name"), Size(255), NotNull]
    public string Name
    {
        get => fields.Name[this];
        set => fields.Name[this] = value;
    }

    [DisplayName("Capacity")]
    public short? Capacity
    {
        get => fields.Capacity[this];
        set => fields.Capacity[this] = value;
    }

    [DisplayName("Status"), NotNull]
    public bool? Status
    {
        get => fields.Status[this];
        set => fields.Status[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field GradeId;
        public StringField Code;
        public StringField Name;
        public Int16Field Capacity;
        public BooleanField Status;

    }
}

File: Modules/ServerTypes/School/GradeRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface GradeRow {
    GradeId?: number;
    Code?: string;
    Name?: string;
    Capacity?: number;
    Status?: boolean;
}

export abstract class GradeRow {
    static readonly idProperty = 'GradeId';
    static readonly nameProperty = 'Code';
    static readonly localTextPrefix = 'School.Grade';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<GradeRow>();
}